src/NuGet.Services.AzureSearch/Owners2AzureSearch/OwnerSetComparer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.AzureSearch.Owners2AzureSearch
+{
+    public class OwnerSetComparer : IOwnerSetComparer
+    {
+        private static readonly string[] EmptyStringArray = new string[0];
+
+        private readonly ILogger _logger;
+
+        public OwnerSetComparer(ILogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public SortedDictionary<string, string[]> Compare(
+            SortedDictionary<string, SortedSet<string>> oldData,
+            SortedDictionary<string, SortedSet<string>> newData)
+        {
+            // We use a very simplistic algorithm here. Perform one pass on the new data to find the added or changed
+            // package IDs. Then perform a second pass on the old data to find removed package IDs. We can optimize
+            // this later if necessary.
+            //
+            // We emit all of the usernames when a package ID's owners have changed instead of the delta. This is
+            // because Azure Search does not have a way to append or remove a specific item from a field that is an
+            // array. The entire new array needs to be provided.
+            var result = new SortedDictionary<string, string[]>(StringComparer.OrdinalIgnoreCase);
+
+            // First pass: find added or changed sets.
+            foreach (var pair in newData)
+            {
+                var id = pair.Key;
+                var newOwners = pair.Value;
+                if (!oldData.TryGetValue(id, out var oldOwners))
```

Comment:
>id [](start = 41, length = 2)

are the ids normalized before this method? I'm thinking about cases when we change capitalization of id by customer request.

Reply:
nope, they are the original casing found in the DB.

Reply:
so might cause a problem, no?

Reply:
`TryGetValue` observes the container's comparer, which is `StringComparer.OrdinalIgnoreCase`. If that doesn't address what you mean, I don't understand.

Reply:
I think @skofman1 is concerned that one set may have normalized IDs while the other doesn't, so comparisons will fail. Given that the comparison is case-insensitive, there isn't any issue here, right?